lib/components/shorts/shorts_video_player_screen.dart

```diff
@@ -0,0 +1,224 @@
+    } else {
+      player = ref.watch(primaryPlayerProvider);
+    }
+    final playerId = player?.playerId ?? '';
+    return Stack(
+      children: [
+        GestureDetector(
+          onTapDown: (details) {
+            if (widget.controller.value.isPlaying) {
+              widget.stopControllerAtIndex(widget.index);
+            } else {
+              widget.playControllerAtIndex(widget.index);
+            }
+            setState(() {
+              isVideoPlaying = widget.controller.value.isPlaying;
+            });
+            return;
+          },
+          child: Align(
+            alignment: Alignment.center,
+            child: AspectRatio(
+              aspectRatio: 16 / 9,
+              child: VideoPlayer(widget.controller),
+            ),
+          ),
+        ),
+        FadingPlayButton(opacityLevel: isVideoPlaying ? 0.0 : 1.0, isVideoPlaying: isVideoPlaying),
+        ForwardButton(widget: widget, ref: ref, playerId: playerId, mounted: mounted),
+        Positioned(
+          right: 30,
+          bottom: 20,
+          child: GestureDetector(
+            behavior: HitTestBehavior.opaque,
+            onTap: () async {
+              setState(() {
+                widget.controller.setVolume(isVideoMuted ? 1.0 : 0.0);
+                isVideoMuted = !isVideoMuted;
+              });
+            },
+            child: Icon(
+              isVideoMuted ? Icons.volume_up_outlined : Icons.volume_off_outlined,
+              size: 40,
+            ),
+          ),
+        ),
+        VideoDescription(
+          episodeId: widget.shortsVideo.id,
+          title: widget.shortsVideo.title,
+          urlLink: '/embed/${widget.shortsVideo.id}?autoplay=true&t=${widget.controller.value.duration.inSeconds.toString()}',
+          playerId: playerId,
+        ),
+        Positioned(
+          left: 30,
+          top: 30,
+          child: IconButton(
+              color: BccmColors.label4,
+              onPressed: () {
+                context.router.pop();
+              },
+              icon: const Icon(
+                Icons.arrow_back_ios_new_outlined,
+                size: 35,
+              )),
+        ),
+      ],
+    );
+  }
+}
+
+class FadingPlayButton extends StatelessWidget {
+  const FadingPlayButton({Key? key, required this.opacityLevel, required this.isVideoPlaying}) : super(key: key);
+
+  final double opacityLevel;
+  final bool isVideoPlaying;
+
+  @override
+  Widget build(BuildContext context) {
+    return Center(
+      child: AnimatedOpacity(
+        opacity: opacityLevel,
+        duration: const Duration(seconds: 2),
+        child: Icon(
+          isVideoPlaying ? Icons.pause_sharp : Icons.play_arrow,
+          size: 50,
+          color: Colors.amber,
+        ),
+      ),
+    );
+  }
+}
+
+class ForwardButton extends StatelessWidget {
+  const ForwardButton({
+    super.key,
+    required this.widget,
+    required this.ref,
+    required this.playerId,
```

Comment:
Don't pass the playerId around, it's a bit confusing especially since you are using a different player system.
Just access it with ref.read() wherever you need it.